cluster-autoscaler/core/scaledown/eligibility/eligibility.go

```diff
@@ -0,0 +1,182 @@
+			c.unremovableNodes.AddReason(node, simulator.RecentlyUnremovable)
+			skipped++
+			continue
+		}
+
+		reason, utilInfo := c.checkNodeUtilization(timestamp, nodeInfo)
+		if utilInfo != nil {
+			utilizationMap[node.Name] = *utilInfo
+		}
+		if reason != simulator.NoReason {
+			c.unremovableNodes.AddReason(node, reason)
+			continue
+		}
+
+		currentlyUnneededNodeNames = append(currentlyUnneededNodeNames, node.Name)
+	}
+
+	if skipped > 0 {
+		klog.V(1).Infof("Scale-down calculation: ignoring %v nodes unremovable in the last %v", skipped, c.context.AutoscalingOptions.UnremovableNodeRecheckTimeout)
+	}
+	return currentlyUnneededNodeNames, utilizationMap
+}
+
+func (c *Checker) checkNodeUtilization(timestamp time.Time, nodeInfo *schedulerframework.NodeInfo) (simulator.UnremovableReason, *utilization.Info) {
+	node := nodeInfo.Node()
+
+	// Skip nodes marked to be deleted, if they were marked recently.
+	// Old-time marked nodes are again eligible for deletion - something went wrong with them
+	// and they have not been deleted.
+	if actuation.IsNodeBeingDeleted(node, timestamp) {
+		klog.V(1).Infof("Skipping %s from delete consideration - the node is currently being deleted", node.Name)
+		return simulator.CurrentlyBeingDeleted, nil
+	}
+
+	// Skip nodes marked with no scale down annotation
+	if HasNoScaleDownAnnotation(node) {
+		klog.V(1).Infof("Skipping %s from delete consideration - the node is marked as no scale down", node.Name)
+		return simulator.ScaleDownDisabledAnnotation, nil
+	}
+
+	utilInfo, err := utilization.Calculate(nodeInfo, c.context.IgnoreDaemonSetsUtilization, c.context.IgnoreMirrorPodsUtilization, c.context.CloudProvider.GPULabel(), timestamp)
+	if err != nil {
+		klog.Warningf("Failed to calculate utilization for %s: %v", node.Name, err)
+	}
+
+	nodeGroup, err := c.context.CloudProvider.NodeGroupForNode(node)
+	if err != nil {
+		klog.Warning("Node group not found for node %v: %v", node.Name, err)
+		return simulator.UnexpectedError, nil
+	}
+	if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
+		// We should never get here as non-autoscaled nodes should not be included in scaleDownCandidates list
+		// (and the default PreFilteringScaleDownNodeProcessor would indeed filter them out).
+		klog.Warningf("Skipped %s from delete consideration - the node is not autoscaled", node.Name)
+		return simulator.NotAutoscaled, nil
+	}
+
+	underutilized, err := c.isNodeBelowUtilizationThreshold(node, nodeGroup, utilInfo)
+	if err != nil {
+		klog.Warningf("Failed to check utilization thresholds for %s: %v", node.Name, err)
+		return simulator.UnexpectedError, nil
+	}
+	if !underutilized {
+		klog.V(4).Infof("Node %s is not suitable for removal - %s utilization too big (%f)", node.Name, utilInfo.ResourceName, utilInfo.Utilization)
+		return simulator.NotUnderutilized, &utilInfo
+	}
+
+	klog.V(4).Infof("Node %s - %s utilization %f", node.Name, utilInfo.ResourceName, utilInfo.Utilization)
```

Comment:
Drive-by fix idea: is this maybe an opportunity to limit the immense spam this line produces in large clusters (klogx?)? This is obviously not in scope of your changes, so feel free to ignore this comment.

Reply:
Seems like a simple change so sure, why not.